### ISSUE TITLE
So many NOT_PAIRED reads leading to no significant results

### ISSUE THREAD

**Issue by CrazyHsu** [Xinglab/rmats-turbo#112]
Hi Eric,
I'm using rMATs(version 4.1.1), but after long time running, it turned out no significant differential AS events. Then I checked results, I found there are two bam files I used for one sample which were found to contain a lot of __NOT_PAIRED__ reads, as below:
```
/Projects/B73_endo/RNA-seq/alignment/repeat0/repeat0.sorted.bam
USED: 25909180
NOT_PAIRED: 55362165
NOT_NH_1: 14209991
NOT_EXPECTED_CIGAR: 127831
NOT_EXPECTED_READ_LENGTH: 0
NOT_EXPECTED_STRAND: 0
EXON_NOT_MATCHED_TO_ANNOTATION: 60557
JUNCTION_NOT_MATCHED_TO_ANNOTATION: 1440566
CLIPPED: 54584159
TOTAL_FOR_BAM: 151694449
/Projects/B73_endo/RNA-seq/alignment/repeat1/repeat1.sorted.bam
USED: 110659
NOT_PAIRED: 83050645
NOT_NH_1: 213621
NOT_EXPECTED_CIGAR: 398
NOT_EXPECTED_READ_LENGTH: 0
NOT_EXPECTED_STRAND: 0
EXON_NOT_MATCHED_TO_ANNOTATION: 473
JUNCTION_NOT_MATCHED_TO_ANNOTATION: 10970
CLIPPED: 44110697
TOTAL_FOR_BAM: 127497463
/Projects/B73_root/RNA-seq/alignment/repeat0/repeat0.sorted.bam


USED: 62426199
NOT_PAIRED: 6943563
NOT_NH_1: 26319865
NOT_EXPECTED_CIGAR: 389749
NOT_EXPECTED_READ_LENGTH: 0
NOT_EXPECTED_STRAND: 0
EXON_NOT_MATCHED_TO_ANNOTATION: 132963
JUNCTION_NOT_MATCHED_TO_ANNOTATION: 3328473
CLIPPED: 54410038
TOTAL_FOR_BAM: 153950850
/Projects/B73_root/RNA-seq/alignment/repeat1/repeat1.sorted.bam
USED: 36969940
NOT_PAIRED: 4027165
NOT_NH_1: 7821769
NOT_EXPECTED_CIGAR: 210644
NOT_EXPECTED_READ_LENGTH: 0
NOT_EXPECTED_STRAND: 0
EXON_NOT_MATCHED_TO_ANNOTATION: 54200
JUNCTION_NOT_MATCHED_TO_ANNOTATION: 1893617
CLIPPED: 26306758
TOTAL_FOR_BAM: 77284093
```
And for the other two bam files I used for comparison, it seems like everything looks normal. Actually, all these bam files were produced through the same process, which used hisat2 as mapper.

I also checked if my first 2 bam files contains so many UN_PAIRED reads with the command `samtools view -c -f 1 /Projects/B73_endo/RNA-seq/alignment/repeat0/repeat0.sorted.bam`, the count number is **151694449** which is consistent with the number in **TOTAL_FOR_BAM: 151694449**. So I'm confused, can you help me? Or should I try rMATs again using STAR as mapper?

**Comment by EricKutschera** [comment]
What rMATS means by `NOT_PAIRED` is that the second flag bit for the alignment is not set. That second bit is described as "each segment properly aligned according to the aligner" (https://samtools.github.io/hts-specs/SAMv1.pdf). `samtools view -c -f 1 {bam}` will check the first flag bit. The first flag bit is described as "template having multiple segments in sequencing"

The samtools command shows that all of the reads were part of a pair and the rMATS `NOT_PAIRED` output indicates that the aligner had some issue when aligning some of those reads as a pair. You could try aligning with STAR to see if it is able to "properly align" more reads

The output also shows that a lot of the reads are not used because they are `CLIPPED`. You could run rMATS with `--allow-clipping` to let rMATS use those reads. It looks like hisat2 has an option to disable clipping `--no-softclip`: http://daehwankimlab.github.io/hisat2/manual/

**Comment by CrazyHsu** [comment]
Hi, @EricKutschera . Thanks for your prompt reply. I will try the `--allow-clipping` parameter and STAR. 
BTW, the species I'm studying is maize, but every run will take 1 day or more, is it normal for rMATs?

**Comment by EricKutschera** [comment]
rMATS runtime depends on the number of input reads and the number of detected events. If there are a lot of reads or events then rMATS could take a long time to run. Using the parameter `--novelSS` allows rMATS to detect more novel events, but it can also increase the runtime by a lot

One way to decrease the runtime is to use `--nthread` to utilize multiple cpu cores. rMATS can use as many threads as there are input bam files. rMATS can also be run across multiple machines by using `--task prep` and `--task post`: https://github.com/Xinglab/rmats-turbo/tree/v4.1.1#running-prep-and-post-separately

**Comment by CrazyHsu** [comment]
Yes, I have set the thread to 40, but it seemed that rMATs only used 2 threads during the process. It may be that I have too many input reads and events.

**Comment by ouyang-lang** [comment]
Hello, I would like to ask, have you compared your ONT data with HisAT2? Is the STAR ratio high? I had the same problem
